src/clients/java/docs.zig

```diff
@@ -0,0 +1,399 @@
+    \\         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    \\  <modelVersion>4.0.0</modelVersion>
+    \\
+    \\  <groupId>api</groupId>
+    \\  <artifactId>api</artifactId>
+    \\  <version>1.0-SNAPSHOT</version>
+    \\
+    \\  <properties>
+    \\    <maven.compiler.source>11</maven.compiler.source>
+    \\    <maven.compiler.target>11</maven.compiler.target>
+    \\  </properties>
+    \\
+    \\  <build>
+    \\    <plugins>
+    \\      <plugin>
+    \\        <groupId>org.apache.maven.plugins</groupId>
+    \\        <artifactId>maven-compiler-plugin</artifactId>
+    \\        <version>3.8.1</version>
+    \\        <configuration>
+    \\          <compilerArgs>
+    \\            <arg>-Xlint:all,-options,-path</arg>
+    \\          </compilerArgs>
+    \\        </configuration>
+    \\      </plugin>
+    \\
+    \\      <plugin>
+    \\        <groupId>org.codehaus.mojo</groupId>
+    \\        <artifactId>exec-maven-plugin</artifactId>
+    \\        <version>1.6.0</version>
+    \\        <configuration>
+    \\          <mainClass>Main</mainClass>
+    \\        </configuration>
+    \\      </plugin>
+    \\    </plugins>
+    \\  </build>
+    \\
+    \\  <dependencies>
+    \\    <dependency>
+    \\      <groupId>com.tigerbeetle</groupId>
+    \\      <artifactId>tigerbeetle-java</artifactId>
+    \\      <version>0.0.1-3258</version>
+    \\    </dependency>
+    \\  </dependencies>
+    \\</project>
+    ,
+
+    .test_file_name = "Main",
+
+    .install_sample_file = 
+    \\package com.tigerbeetle.examples;
+    \\
+    \\import com.tigerbeetle.*;
+    \\
+    \\public final class Main {
+    \\    public static void main(String[] args) {
+    \\        System.out.println("Import ok!");
+    \\    }
+    \\}
+    ,
+
+    .install_prereqs = "apk add -U maven openjdk11",
+
+    .install_commands = 
+    \\mvn install
+    ,
+
+    .install_sample_file_build_commands = 
+    \\mvn compile
+    ,
+    .install_sample_file_test_commands = "mvn exec:java",
+
+    .install_documentation = "",
+
+    .examples = "",
+
+    .client_object_example = 
+    \\Client client = new Client(
+    \\  0,
+    \\  new String[] {"127.0.0.1:3001", "127.0.0.1:3002", "127.0.0.1:3003"}
+    \\);
+    ,
+
+    .client_object_documentation = 
+    \\If you create a `Client` like this, don't forget to call
+    \\`client.close()` when you are done with it. Otherwise you
+    \\can use the
+    \\[try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html)
+    \\syntax:
+    \\```java
+    \\try (var client = new Client(...)) {
+    \\  // Use client
+    \\} catch (Exception e) {
+    \\  // Handle exception
+    \\}
+    \\```
+    ,
+
+    .create_accounts_example = 
+    \\AccountBatch accounts = new AccountBatch(1);
+    \\accounts.add();
+    \\accounts.setId(UInt128.asBytes(new java.math.BigInteger("137")));
```

Comment:
That's really a good example: I think it's not the way we should encourage using 128-bit IDs in Java, because we are heap allocating three objects (and because it's so verbose!).

The intended usage of `BigInteger` in our API is for handling numeric literals that cannot be expressed as a Java signed 64-bit long, like this:

In Zig:
```zig
assert(@bitCast(u128, [2]i64{ -100, -1000 }) == 340282366920938445035077277795926146972);
```

In Java:
```java
final var value = UInt128.asBigInteger(-100, -1000);
final var expected = new BigInteger("340282366920938445035077277795926146972");
assertEquals(expected, value);
```

**Suggestions: **

1. As long `"137"` is can be expressed as a Java `long` we don't need a `BigInteger` for that, so we can just do `accounts.setId(UInt128.asBytes(137))`, although we're still allocating a byte array.

2. We can express a 128-bit integer using two `long`, so `accounts.setId(137, 0)` is the preferable usage here, no heap allocation is made. I agree that this second argument `0` can be somehow confusing, especially in an introductory example.

3. We can introduce a new overload for all 128-bit setters expecting only one `long` argument, making it as simple as `accounts.setId(137)`.

Reply:
Good suggestions! I've added 3 to java_bindings.zig. I'll swap the examples over.